### PR TITLE
deno 1.36.1

### PR DESCRIPTION
**Pull request by ZhongRuoyu**
Created with `brew bump-formula-pr`.

- [x] `resource` blocks have been checked for updates.

### CODE REVIEW

**Comment by ZhongRuoyu** [comment]
```
    --- stderr
    thread 'main' panicked at 'assertion failed: Command::new(gn()).arg(format!(\"--script-executable={}\",\n                                python())).arg(\"args\").arg(gn_out_dir).arg(\"--list\").status().unwrap().success()', /github/home/.cache/Homebrew/cargo_cache/registry/src/index.crates.io-6f17d22bba15001f/v8-0.74.3/build.rs:229:3
    note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
  error: failed to compile `deno v1.36.1 (/tmp/deno-20230810-11028-f3fz7o/deno/cli)`, intermediate artifacts can be found at `/tmp/deno-20230810-11028-f3fz7o/deno/target`
```

Same as in #136476 and #138431.

**Comment by ZhongRuoyu** [comment]
```
  [libz-ng-sys 1.1.12] thread 'main' panicked at '
  [libz-ng-sys 1.1.12] failed to execute command: No such file or directory (os error 2)
  [libz-ng-sys 1.1.12] is `cmake` not installed?
  [libz-ng-sys 1.1.12] 
  [libz-ng-sys 1.1.12] build script failed, must exit now', /github/home/.cache/Homebrew/cargo_cache/registry/src/index.crates.io-6f17d22bba15001f/cmake-0.1.50/src/lib.rs:1098:5
  [libz-ng-sys 1.1.12] note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
  error: failed to run custom build command for `libz-ng-sys v1.1.12`
```

([Logs](https://github.com/Homebrew/homebrew-core/actions/runs/5821166566/job/15824136290?pr=139193#step:4:18213))

File: Formula/deno.rb
Comment (ZhongRuoyu) [inline Formula/deno.rb:1]: I've tested these changes locally, but this needs a rebase against #139382 to build.

File: Formula/deno.rb [hunk 1]
```diff
@@ -73,6 +80,17 @@ def install
...
+    # Avoid vendored dependencies.
+    inreplace "ext/ffi/Cargo.toml",
+              /^libffi-sys = "(.+)"$/,
+              'libffi-sys = { version = "\\1", features = ["system"] }'
+    inreplace "ext/node/Cargo.toml",
+              /^libz-sys = { version = "(.+)", features = \["static"\] }$/,
+              'libz-sys = "\\1"'
+    # inreplace "Cargo.toml",
+    #           /^rusqlite = { version = "(.+)", features = \["unlock_notify", "bundled"\] }$/,
+    #           'rusqlite = { version = "\\1", features = ["unlock_notify"] }'
```
Comment (ZhongRuoyu) [inline Formula/deno.rb:92]: Commented this out to unblock the release. Will revisit later.

**Review by branchv — APPROVED** [review]

**Pull request merged**